engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -110,10 +120,15 @@ class NextgenEngine @JvmOverloads constructor(
                         .map { (field, service) ->
                             async {
                                 try {
-                                    executeTopLevelField(field, service, executionContext)
+                                    val resolvedService = fieldToService.resolveDynamicService(field, service)
+
+                                    executeTopLevelField(field, resolvedService, executionContext)
                                 } catch (e: Throwable) {
                                     when (e) {
-                                        is GraphQLError -> newExecutionResult(error = e)
+                                        is GraphQLError -> newExecutionResult(
+                                            error = e,
+                                            data = mutableMapOf(field.resultKey to null)
```

Comment:
This changes general behaviour, not only dynamic resolution related stuff. But I think is for the better. It will ensure a `field: null` property on the data is added when there is a big error that prevented the actual execution from happening.

It didn't break any existing tests too.

Reply:
Yup that sounds good.